server/service/src/invoice_line/inbound_shipment_line/zero_line_quantity/generate.rs

```diff
@@ -0,0 +1,52 @@
+use repository::{InvoiceLineRow, InvoiceRow, InvoiceRowStatus};
+
+pub struct GenerateResult {
+    pub invoice_row: InvoiceRow,
+    pub new_line: InvoiceLineRow,
+    pub stock_line_id: Option<String>,
+}
+
+pub fn generate(
+    user_id: &str,
+    existing_invoice_row: InvoiceRow,
+    existing_invoice_line: InvoiceLineRow,
+) -> GenerateResult {
+    let new_line = generate_line(existing_invoice_line.clone());
+
+    let invoice_row = InvoiceRow {
+        status: InvoiceRowStatus::Verified,
+        delivered_datetime: Some(chrono::Utc::now().naive_utc()),
+        verified_datetime: Some(chrono::Utc::now().naive_utc()),
+        user_id: if existing_invoice_row.user_id != Some(user_id.to_string()) {
+            Some(user_id.to_string())
+        } else {
+            existing_invoice_row.user_id
+        },
+        ..existing_invoice_row
+    };
+
+    GenerateResult {
+        invoice_row,
+        new_line,
+        stock_line_id: existing_invoice_line.stock_line_id,
+    }
+}
+
+fn generate_line(line: InvoiceLineRow) -> InvoiceLineRow {
+    InvoiceLineRow {
```

Comment:
Overkill None'ing?

Reply:
I mean.. if you haven't received it or it is not valid anymore, is it wrong? Since a transfer would send batch info and other things as well...

Reply:
But the information might still be valuable, i.e. what has been sent etc. And then you just said you have received 0 of these. I guess that's why mSupply has a "send" and a "received" field...